CMakeLists.txt

```
@@ -544,6 +544,22 @@ if(EXISTS ${APPLICATION_SOURCE_DIR}/VERSION)
   add_dependencies(zephyr_interface app_version_h)
 endif()
 
+# Create the kernel library now so that other parts of the system can use
+# it with target_link_libraries or other commands.
+#
+# If a pre-built static library containing kernel code exists in the kernel
+# directory, kernel/libkernel.a, link it with the application code instead
+# of building from source.
+
+set(libkernel ${CMAKE_CURRENT_SOURCE_DIR}/kernel/libkernel${CMAKE_STATIC_LIBRARY_SUFFIX})
+if(EXISTS ${libkernel})
+  add_library(kernel INTERFACE)
+  set_property(TARGET kernel PROPERTY KERNEL_INTERFACE TRUE)
```

thanks for this.

Let's use the opportunity to import pre-built libs the proper way:
```suggestion
  add_library(kernel STATIC IMPORTED GLOBAL)
  set_property(TARGET kernel PROPERTY IMPORTED_LOCATION ${libkernel})
```